airflow/contrib/operators/kubernetes_pod_operator.py

```diff
@@ -23,6 +25,28 @@
 from airflow.utils.state import State
 
 
+def create_labels_for_pod(context):
+    """
+    Generate labels for the pod s.t. we can track it in case of Operator crash
+    :param context:
+    :return:
+    """
+    labels = {
+        'dag_id': context['dag'].dag_id,
+        'task_id': context['task'].task_id,
+        'exec_date': context['ts']
+    }
+    # In the case of sub dags this is just useful
+    if context['dag'].parent_dag:
+        labels['parent_dag_id'] = context['dag'].parent_dag.dag_id
+    # Replace unsupported characters with dashes & trim at 63 chars (max allowed)
+    for label_id, label in labels.items():
+        label = re.sub(r'[^-A-Za-z0-9_.]+', '-', label)[:63]
```

Comment:
We refactor `_make_safe_label_value` from KubeExcutor and move it to `make_safe_label_value` on PodGenerator then call it form both places.

Reply:
Agreed